fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -319,14 +318,24 @@ private int getChildrenLimit() {
     @DELETE
     @Timed
     public Response deleteObject() {
-        evaluateRequestPreconditions(request, servletResponse, resource(), session);
+        final FedoraResource resource = resource();
+        if (resource instanceof ContainerImpl) {
+            final String depth = headers.getHeaderString("Depth");
+            System.out.println(depth);
+            if (depth != null && !depth.equalsIgnoreCase("infinity")) {
+                throw new ClientErrorException("Depth header, if present, must be set to 'infinity' for containers",
+                        SC_BAD_REQUEST);
+            }
+        }
+
+        evaluateRequestPreconditions(request, servletResponse, resource, session);
 
         LOGGER.info("Delete resource '{}'", externalPath);
 
-        final AcquiredLock lock = lockManager.lockForDelete(resource().getPath());
```

Comment:
Actually, probably leave it as `resource()` for consistency sake with everything else in this class.  The implementation looks like it simply retrieves a field anyway (i.e. only does a lookup once).

Reply:
Ok